lib/xml-name-validator.js

```diff
@@ -1,17 +1,9 @@
 "use strict";
-const parser = require("./generated-parser.js");
 
 exports.name = potentialName => {
-  return mapResult(parser.startWith("Name").exec(potentialName));
+  return /^[:A-Z_a-z\u00C0-\u00D6\u00D8-\u00F6\u00F8-\u02FF\u0370-\u037D\u037F-\u1FFF\u200C-\u200D\u2070-\u218F\u2C00-\u2FEF\u3001-\uD7FF\uF900-\uFDCF\uFDF0-\uFFFD\uD800-\uDB7F\uDC00-\uDFFF][:A-Z_a-z\u00C0-\u00D6\u00D8-\u00F6\u00F8-\u02FF\u0370-\u037D\u037F-\u1FFF\u200C-\u200D\u2070-\u218F\u2C00-\u2FEF\u3001-\uD7FF\uF900-\uFDCF\uFDF0-\uFFFD\uD800-\uDB7F\uDC00-\uDFFF\-.0-9\u00B7\u0300-\u036F\u203F-\u2040]*$/u.test(potentialName);
 };
 
 exports.qname = potentialQname => {
-  return mapResult(parser.startWith("QName").exec(potentialQname));
+  return /(?:^[A-Z_a-z\u00C0-\u00D6\u00D8-\u00F6\u00F8-\u02FF\u0370-\u037D\u037F-\u1FFF\u200C-\u200D\u2070-\u218F\u2C00-\u2FEF\u3001-\uD7FF\uF900-\uFDCF\uFDF0-\uFFFD\uD800-\uDB7F\uDC00-\uDFFF][A-Z_a-z\u00C0-\u00D6\u00D8-\u00F6\u00F8-\u02FF\u0370-\u037D\u037F-\u1FFF\u200C-\u200D\u2070-\u218F\u2C00-\u2FEF\u3001-\uD7FF\uF900-\uFDCF\uFDF0-\uFFFD\uD800-\uDB7F\uDC00-\uDFFF\-.0-9\u00B7\u0300-\u036F\u203F-\u2040]*:[A-Z_a-z\u00C0-\u00D6\u00D8-\u00F6\u00F8-\u02FF\u0370-\u037D\u037F-\u1FFF\u200C-\u200D\u2070-\u218F\u2C00-\u2FEF\u3001-\uD7FF\uF900-\uFDCF\uFDF0-\uFFFD\uD800-\uDB7F\uDC00-\uDFFF][A-Z_a-z\u00C0-\u00D6\u00D8-\u00F6\u00F8-\u02FF\u0370-\u037D\u037F-\u1FFF\u200C-\u200D\u2070-\u218F\u2C00-\u2FEF\u3001-\uD7FF\uF900-\uFDCF\uFDF0-\uFFFD\uD800-\uDB7F\uDC00-\uDFFF\-.0-9\u00B7\u0300-\u036F\u203F-\u2040]*$)|(?:^[A-Z_a-z\u00C0-\u00D6\u00D8-\u00F6\u00F8-\u02FF\u0370-\u037D\u037F-\u1FFF\u200C-\u200D\u2070-\u218F\u2C00-\u2FEF\u3001-\uD7FF\uF900-\uFDCF\uFDF0-\uFFFD\uD800-\uDB7F\uDC00-\uDFFF][A-Z_a-z\u00C0-\u00D6\u00D8-\u00F6\u00F8-\u02FF\u0370-\u037D\u037F-\u1FFF\u200C-\u200D\u2070-\u218F\u2C00-\u2FEF\u3001-\uD7FF\uF900-\uFDCF\uFDF0-\uFFFD\uD800-\uDB7F\uDC00-\uDFFF\-.0-9\u00B7\u0300-\u036F\u203F-\u2040]*$)/u.test(potentialQname);
```

Comment:
Are attributes with a missing prefix (like `:attr`) allowed?

Reply:
No, it's either `NCName:NCName` or `NCName`.

Reply:
I think we can also move the $ on both of the | branches to the actual regex end?

Reply:
If we do that we need another `(?: ... )` around the whole thing apparently: try `^(?:(?:[a-z][a-z]*)|(?:[a-z][a-z]*:[a-z][a-z]*))$` vs. `^(?:[a-z][a-z]*)|(?:[a-z][a-z]*:[a-z][a-z]*)$` on https://www.regextester.com/

Reply:
> No, it's either `NCName:NCName` or `NCName`.

Ah I misread the regex. Overlooked the fact we were requiring a char before the char* construct.

> If we do that we need another (?: ... ) around the whole thing apparently

Ah yeah, the | is screwing us there. Doubt that's worth it then.